compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -462,7 +462,21 @@ pub fn make_format_args(
             let msg = if let FormatArgumentKind::Named(_) = args.explicit_args()[i].kind {
                 "named argument never used"
             } else {
-                "argument never used"
+                let mut err = "argument never used";
+                match args.explicit_args()[i].expr.to_ty() {
+                    Some(expr) => match expr.kind.is_simple_path() {
+                        Some(symbol) => {
+                            let current_arg = symbol.as_str();
+                            let current_arg_ph = "{".to_owned() + current_arg + "}";
+                            if current_arg.len() > 0 && fmt_str.contains(current_arg_ph.as_str()) {
+                                err = "argument never used, consider removing it"
```

Comment:
since the argument is also a duplicate of an existing argument, this should probably mention that the argument is a duplicate of an inline argument, which is the reason it should be removed.

Reply:
"argument is a duplicate of an inline argument" is an intuitive error suggestion I think? or should I include ", consider removing it" afterwards?